src/jit/gentree.h

```diff
@@ -2399,12 +2399,23 @@ struct ReturnTypeDesc
 public:
     ReturnTypeDesc()
     {
+#ifdef _TARGET_X86_
+        m_regType0 = TYP_INT;
+        m_regType1 = TYP_INT;
+
+#ifdef DEBUG
+        m_inited = true;
+#endif
+
+#else
```

Comment:
How about adding a function to reset the state of ReturnTypeDesc to un-initialized state?

e.g. 

```
void Reset()
{
   m_regType0 = TYP_UNKNOWN;
   m_regType1 = TYP_UNKNOWN;

#ifdef DEBUG
     m_inited = false;
#endif
}
```

Now wherever we need to re-initialize (e.g. fgMorphIntoHelperCall())
retTypeDesc->Reset();
retTypeDesc->Initialize()

Note that the default constructor of ReturnTypeDesc can also use Reset() to 

e.g.

```
ReturnTypeDesc()
{
   Reset();
}
```